Sources/FuturedHelpers/Helpers/CoordinatorSceneFlowProvider.swift

```diff
@@ -0,0 +1,210 @@
+//
+//  CoordinatorSceneFlowProvider.swift
+//
+//  Created by Simon Sestak on 31/07/2024.
+//
+
+import SwiftUI
+
+/** A protocol to represent navigable destinations in a reusable scene flow
+ ```
+ protocol TemplateFlowDestination: CoordinatorSceneFlowDestination {
+    static var destination: Self { get }
+    static var end: Self { get }
+ }
+ ```
+ **/
+protocol CoordinatorSceneFlowDestination: Hashable, Identifiable, Equatable {}
+
+/**
+ A protocol defining an interface for reusable scene flow providers.
+ ```
+final class TemplateSceneFlowProvider: CoordinatorSceneFlowProvider {
+     let onNavigateToDestination: (Destination) -> Void
+     let onPop: () -> Void
+     let onPresentSheet: ((Destination) -> Void)? = nil
+     let onDismissSheet: (() -> Void)? = nil
+     let onPresentFullscreenCover: ((Destination) -> Void)? = nil
+     let onDismissFullscreenCover: (() -> Void)? = nil
+     let onPopToDestination: ((Destination?) -> Void)? = nil
+     let onShowError: ((AppError) -> Void)? = nil
+
+     private let container: BasicCapitalContainer
+
+     init(container: BasicCapitalContainer, onNavigateToDestination: @escaping (Destination) -> Void, onPop: @escaping () -> Void) {
+         self.container = container
+         self.onNavigateToDestination = onNavigateToDestination
+         self.onPop = onPop
+     }
+
+     static func rootView(with instance: TemplateSceneFlowProvider) -> some View {
+         LearnComponent(
+            model: LearnComponentModel(
+                dataCache: instance.container.dataCache
+            ) { [weak instance] event in
+                switch event {
+                case .showUnderstandBasicCapital:
+                    instance?.navigate(to: .destination)
+                }
+            }
+         )
+     }
+
+     func scene(for destination: Destination) -> some View {
+         switch destination {
+         case .destination:
+             Color.red
+         case .end:
+             EmptyView()
+         }
+     }
+
+     enum Destination: String, TemplateFlowDestination {
+         case destination
+         case end
+
+         var id: String {
+             rawValue
+         }
+     }
+ }
+
+final class TemplateCoordinator: NavigationStackCoordinator {
+    @Published var path: [Destination] = []
+    @Published var sheet: Destination?
+    @Published var fullscreenCover: Destination?
+    @Published var alertModel: AlertModel?
+
+    private let container: BasicCapitalContainer
+
+    private lazy var templateSceneProvider: TemplateSceneFlowProvider = {
+        TemplateSceneFlowProvider(container: container, onNavigateToDestination: { [weak self] destination in
+            if destination == .end {
+                self?.navigate(to: .flowSpecificDestinationAfterEmbededFlow)
+            } else {
+                self?.navigate(to: destination)
+            }
+        }, onPop: { [weak self] in
+            self?.pop()
+        })
```

Comment:
This is a matter of opinion, but I prefer following multi-closure function calls:
```suggestion
        TemplateSceneFlowProvider(
            container: container, 
            onNavigateToDestination: { [weak self] destination in
                if destination == .end {
                    self?.navigate(to: .flowSpecificDestinationAfterEmbededFlow)
                } else {
                    self?.navigate(to: destination)
                }
            },
            onPop: { [weak self] in
                self?.pop()
            }
        )
```